api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/marxan-directory.service.ts

```diff
@@ -0,0 +1,34 @@
+import { join as joinPath } from 'path';
+import { Injectable } from '@nestjs/common';
+
+import { Workspace } from '../ports/workspace';
+
+import { FileReader } from './file-reader';
+
+@Injectable()
+export class MarxanDirectory {
+  constructor(private readonly reader: FileReader) {}
+
+  get(
+    type: 'INPUTDIR' | 'OUTPUTDIR',
+    within: Workspace,
```

Comment:
I think `Workspace` is not entirely required here, maybe consider passing just the `workingDirectory`?

Reply:
It could but the intention was to provide this via Workspace to ensure it is what it is supposed to be instead raw string.

Not strongly convinced to either option tho. Do you see any strong benefits by passing only path?
@Dyostiq

Reply:
From my point of view, it violates the principle of least knowledge a bit and makes the code harder to test as you need to pass the entire workspace instead of only the part. Then, the big object propagates through the stack just only to pass the path. If you think that passing a raw string is not as expressive as you'd like it to be, maybe wrapping it in an object is a solution – but still, you can pass only really needed data.

Reply:
https://github.com/Vizzuality/marxan-cloud/pull/316/commits/9e4b34bef2825e79f15971ea147f78223049ec4c